charts/karpenter/templates/clusterrole.yaml

```diff
@@ -37,5 +37,27 @@ rules:
     resources: ["daemonsets"]
     verbs: ["list", "watch"]
   - apiGroups: ["admissionregistration.k8s.io"]
-    resources: ["validatingwebhookconfigurations", "mutatingwebhookconfigurations"]
-    verbs: ["get", "watch", "list", "update"]
+    resources: ["mutatingwebhookconfigurations", "validatingwebhookconfigurations"]
+    verbs: ["list", "watch"]
+  - apiGroups: ["admissionregistration.k8s.io"]
+    resources: ["mutatingwebhookconfigurations"]
+    verbs: ["get", "update", "delete"]
+    resourceNames: ["defaulting.webhook.provisioners.karpenter.sh"]
+  - apiGroups: ["admissionregistration.k8s.io"]
+    resources: ["validatingwebhookconfigurations"]
+    verbs: ["get", "update", "delete"]
+    resourceNames: ["validation.webhook.provisioners.karpenter.sh", "validation.webhook.config.karpenter.sh"]
+  - apiGroups: [""]
```

Comment:
Why is this necessary?

Reply:
see https://github.com/aws/karpenter/issues/1652#issuecomment-1093939872

w/o this, errors will be thrown:

```
2022-04-09T11:24:36.021Z	ERROR	webhook.DefaultingWebhook	Reconcile error	{"commit": "8280611", "knative.dev/traceid": "e7978e69-0230-4a80-acc2-56f67a219715", "knative.dev/key": "defaulting.webhook.provisioners.karpenter.sh", "duration": "104.965499ms", "error": "failed to update webhook: mutatingwebhookconfigurations.admissionregistration.k8s.io \"defaulting.webhook.provisioners.karpenter.sh\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>"}
2022-04-09T11:24:36.059Z	ERROR	webhook.ValidationWebhook	Reconcile error	{"commit": "8280611", "knative.dev/traceid": "99720244-ddfd-4dca-aff4-fb7e1127a242", "knative.dev/key": "validation.webhook.provisioners.karpenter.sh", "duration": "143.743259ms", "error": "failed to update webhook: validatingwebhookconfigurations.admissionregistration.k8s.io \"validation.webhook.provisioners.karpenter.sh\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>"}
2022-04-09T11:24:36.061Z	ERROR	webhook.ConfigMapWebhook	Reconcile error	{"commit": "8280611", "knative.dev/traceid": "7fcefebf-d579-436c-a48a-a95e8bb3b18d", "knative.dev/key": "validation.webhook.config.karpenter.sh", "duration": "143.589552ms", "error": "failed to update webhook: validatingwebhookconfigurations.admissionregistration.k8s.io \"validation.webhook.config.karpenter.sh\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>"}
```

Reply:
On top, update permission is need, because karpenter wants to set the cabundle property.

Reply:
I'm talking specifically about this one

```
  - apiGroups: [""]
     resources: ["namespaces/finalizers"]
     verbs: ["update"]
```
Why do we need to update finalizers on all namespaces?

Reply:
https://github.com/aws/karpenter/pull/1653/files#diff-655e24097d7cb14b02c24694eb19f56a08a5ae6f811e035f911f5211924a22fcR51-R54

Not on all namespaces. The RBAC is limited to `resourceNames: ["{{ .Release.Namespace }}"]`.

Reply:
Looks like this is useful to cleanup webhooks on uninstall. Makes sense to me if we restrict to Release.Namespace.